api/types/role.go

```diff
@@ -1182,7 +1182,7 @@ func (r *RoleV6) CheckAndSetDefaults() error {
 		if err := validateRoleSpecKubeResources(r.Version, r.Spec); err != nil {
 			return trace.Wrap(err)
 		}
-	case V7:
+	case V7, V8:
```

Comment:
The V8 role should have a dedicated validation path. We plan to remove pod, secret, and similar resources in favor of a more flexible `(apiName, objectName)` pair.  This means we need to convert the object kind to its (api,rsourceName) for simplification.

Additionally, we need to determine our approach to support:  
- Whether new Kubernetes resources should be exclusive to role V8, with V8 not accepting `kubernetes_groups/users`.  
- Or, if new Kubernetes resources should be supported by roles V7, V6, etc., while only roles older than V8 continue to support `kubernetes_groups/users`.

Reply:
This PR only bumps the role version without any changes. Additional validation / behavior change will come in future PRs.

kubernetes_resources will not change for v7, v6 etc.

Reply:
added a TODO to make sure it gets done when we have logic changes.